peer.go

```diff
@@ -1169,7 +1176,7 @@ out:
 			}
 
 		case <-p.quit:
-			exitErr = errors.Errorf("peer exiting")
+			exitErr = ErrPeerExiting
```

Comment:
If we are worried about deadlocking waiting for a recv on the `errChan`, maybe we could drain the `sendQueue` on exit, sending errors for all messages?